File: src/main/java/com/freya02/botcommands/application/context/ContextCommandParameter.java
package com.freya02.botcommands.application.context;

import com.freya02.botcommands.application.ApplicationCommandParameter;

public class ContextCommandParameter<T> extends ApplicationCommandParameter<T> {
	public ContextCommandParameter(Class<T> resolverType, Class<?> type) {
		super(resolverType, type);
	}
}

File: src/main/kotlin/com/freya02/botcommands/internal/commands/application/context/ContextCommandParameter.kt
package com.freya02.botcommands.internal.commands.application.context

import com.freya02.botcommands.api.commands.application.builder.ApplicationCommandOptionAggregateBuilder
import com.freya02.botcommands.internal.core.BContextImpl
import com.freya02.botcommands.internal.core.reflection.toAggregatorFunction
import com.freya02.botcommands.internal.parameters.IAggregatedParameter
import com.freya02.botcommands.internal.parameters.MethodParameterImpl

abstract class ContextCommandParameter(
    context: BContextImpl,
    optionAggregateBuilder: ApplicationCommandOptionAggregateBuilder<*>
) : MethodParameterImpl(optionAggregateBuilder.parameter), IAggregatedParameter {
    final override val aggregator = optionAggregateBuilder.aggregator.toAggregatorFunction(context)
}